### PR TITLE
E2E tests using root contracts

### LINKED ISSUE #52: Update ui-e2e to work with contracts at 2.3.0-rc0
Currently they only work with with ~~`2.1.0`~~ `2.2.0` as of #54

### LINKED ISSUE #53: Update oracle-e2e to work with contracts at 2.3.0-rc0
Currently they only work with with `2.2.0`

### PR DESCRIPTION
**Pull request by rzadp**
- Fixes #53
- Fixes #52 
- Progresses #21

#### Changes:
- Modified `Dockerfile.e2e` to utilize docker build cache better
- Change root `contracts` from `2.3.0-rc0` to `2.3.1`
- Removed now-reduntant `contracts-2.2.0`
- Updated `oracle-e2e` to work with root `contracts`
  - Updated `deployERC20.js` script
  - Updated `.env's`
- Updated `ui-e2e` to work with root `contracts`
  - Updated `deployERC20.js` script
  - Updated `.env's`

### CODE REVIEW

**Comment by rzadp** [comment]
@akolotov @patitonar I have a hard time trying to fix the `oracle-e2e` tests when switching from `2.2.0` contracts to `2.3.0-rc0`.

I have updated the `.env's` and the deployment is successful.
The failing tests are `erc to erc` tests, failed assertion is that the erc677 balance of the user is zero.
I have tried switching `ERC20_EXTENDED_BY_ERC677` as `true` and `false` to no effect.

Maybe you will have some tips on how to approach this?

**Comment by akolotov** [comment]
1. I would suggest to start using the version [2.3.0](https://github.com/poanetwork/poa-bridge-contracts/releases/tag/2.3.0) of the contracts that we released tonight.

2. I see that you changed `oracle-e2e/scripts/deployERC20.js`. Could you confirm that token is deployed correctly and minted.

**Comment by patitonar** [comment]
Running `docker logs oracle-e2e_bridge-erc_run_3` while e2e test are running it seems that affirmation-request watcher is being stop because is not a validator when trying to process the transfer
```
{"level":20,"time":1558633217135,"msg":"Processing 1 Transfer events","validator":"0xaaB52d66283F7A1D5978bcFcB55721ACB467384b","name":"watcher-erc-erc-affirmation-request","v":1}
{"level":30,"time":1558633217136,"msg":"Processing transfer 0xe6f4292212b8abc446ea39b32048c973af7c38536559f13535bf2d53239967be","validator":"0xaaB52d66283F7A1D5978bcFcB55721ACB467384b","name":"watcher-erc-erc-affirmation-request","eventTransactionHash":"0xe6f4292212b8abc446ea39b32048c973af7c38536559f13535bf2d53239967be","from":"0xbb140FbA6242a1c3887A7823F7750a73101383e3","value":"10000000000000000","v":1}
{"level":20,"time":1558633217136,"msg":"Estimate gas","validator":"0xaaB52d66283F7A1D5978bcFcB55721ACB467384b","name":"watcher-erc-erc-affirmation-request","eventTransactionHash":"0xe6f4292212b8abc446ea39b32048c973af7c38536559f13535bf2d53239967be","v":1}
{"level":20,"time":1558633217146,"msg":"Check if minimum number of validations was already reached","validator":"0xaaB52d66283F7A1D5978bcFcB55721ACB467384b","name":"watcher-erc-erc-affirmation-request","module":"processAffirmationRequests:estimateGas","v":1}
{"level":20,"time":1558633217151,"msg":"Check if the message was already signed","validator":"0xaaB52d66283F7A1D5978bcFcB55721ACB467384b","name":"watcher-erc-erc-affirmation-request","module":"processAffirmationRequests:estimateGas","v":1}
{"level":20,"time":1558633217153,"msg":"Check if address is a validator","validator":"0xaaB52d66283F7A1D5978bcFcB55721ACB467384b","name":"watcher-erc-erc-affirmation-request","module":"processAffirmationRequests:estimateGas","v":1}
{"level":60,"time":1558633217156,"msg":"Invalid validator","validator":"0xaaB52d66283F7A1D5978bcFcB55721ACB467384b","name":"watcher-erc-erc-affirmation-request","eventTransactionHash":"0xe6f4292212b8abc446ea39b32048c973af7c38536559f13535bf2d53239967be","address":"0xaaB52d66283F7A1D5978bcFcB55721ACB467384b","v":1}
npm ERR! code ELIFECYCLE
npm ERR! errno 10
npm ERR! oracle@1.0.0 watcher:affirmation-request: `./scripts/start-worker.sh watcher affirmation-request-watcher`
npm ERR! Exit status 10
npm ERR! 
npm ERR! Failed at the oracle@1.0.0 watcher:affirmation-request script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-05-23T17_40_17_165Z-debug.log
```

**Comment by patitonar** [comment]
Found the cause of the issue. There is a bug on the deploy script on Home side of ERC_TO_ERC mode. I'll open a PR on the contract side to fix it

**Comment by akolotov** [comment]
> Found the cause of the issue. There is a bug on the deploy script on Home side of ERC_TO_ERC mode. I'll open a PR on the contract side to fix it

The fix is in the up-issue of the contracts: [`2.3.1`](https://github.com/poanetwork/poa-bridge-contracts/releases/tag/2.3.1)

**Comment by rzadp** [comment]
The `2.3.1` contracts fixed the issue.
Newest commits include changing UI-E2E tests as well to use main `contracts`

File: oracle-e2e/envs/contracts-deploy.env [hunk 1]
```diff
@@ -4,6 +4,8 @@ DEPLOYMENT_GAS_LIMIT=4600000
 HOME_DEPLOYMENT_GAS_PRICE=10000000000
 FOREIGN_DEPLOYMENT_GAS_PRICE=10000000000
 GET_RECEIPT_INTERVAL_IN_MILLISECONDS=50
+DEPLOYMENT_GAS_LIMIT_EXTRA=0.2
+DEPLOY_REWARDABLE_TOKEN=false
```
Comment (akolotov) [inline oracle-e2e/envs/contracts-deploy.env]: `DEPLOYMENT_GAS_LIMIT` was replaced by `DEPLOYMENT_GAS_LIMIT_EXTRA` so it is not needed any more.

Please correct this in other `.env` files.

Comment (rzadp) [reply oracle-e2e/envs/contracts-deploy.env]: Done.

**Pull request merged**